Tested this branch locally — works great for `next build`/prod, but it breaks `next dev`: any page rendering `<TinaMarkdown>` 500s.

**Cause:** `TINACMS_MDX_SHIM` is an absolute path. Webpack's `resolve.alias` accepts absolute paths, but Turbopack's `resolveAlias` treats the value as relative — it turns `/Users/.../lib/tinacms-mdx-shim.js` into `./Users/.../lib/tinacms-mdx-shim.js` and can't resolve it:

```
Module not found: Can't resolve './Users/.../lib/tinacms-mdx-shim.js'
GET /events/ai-for-business-leaders 500
```

So the webpack alias (prod, CI build, your bundle measurements) is fine — only `next dev` is affected, which is why CI didn't catch it (CI only runs `next build`, never `next dev`).

**Repro:**
```bash
git checkout perf/tina-mdx-sanitize-url
pnpm dev
curl -s -o /dev/null -w "%{http_code}\n" http://localhost:3000/events/ai-for-business-leaders
# -> 500  (dev log: Can't resolve './Users/.../lib/tinacms-mdx-shim.js')
```

**Fix** — point just the Turbopack alias at a project-relative path; webpack keeps the absolute one:
```js
turbopack: {
  resolveAlias: { "@tinacms/mdx": "./lib/tinacms-mdx-shim.js" },
}
```
Confirmed: with that one change the same page returns 200 and rich text renders. Nothing else needed touching — the shim's `sanitizeUrl` is byte-identical to the installed `@tinacms/mdx@2.1.4` and the `dd3583a7` chunk is gone in prod as you measured.

--- a/next.config.mjs
+++ b/next.config.mjs
@@ -4,11 +4,16 @@ import path from "node:path";
 import { fileURLToPath } from "node:url";
 const withNextPluginPreval = createNextPluginPreval();
 
-// Absolute so webpack and Turbopack resolve it identically regardless of cwd.
-const TINACMS_MDX_SHIM = path.join(
+// The two bundlers want this path in different forms, so don't unify them:
+// webpack's resolve.alias needs an absolute path, while Turbopack's resolveAlias
+// resolves values against the project root and prepends "./" to whatever it is
+// given — an absolute path there becomes "./Users/..." and fails to resolve,
+// 500ing every page that renders <TinaMarkdown> under `next dev`.
+const TINACMS_MDX_SHIM_ABS = path.join(
   path.dirname(fileURLToPath(import.meta.url)),
   "lib/tinacms-mdx-shim.js"
 );
+const TINACMS_MDX_SHIM_REL = "./lib/tinacms-mdx-shim.js";
 
 /** @type {import('next').NextConfig} */
 const config = {
@@ -73,7 +78,7 @@ const config = {
     // Remove once tinacms#7233 lands a `@tinacms/mdx/sanitize-url` subpath.
     config.resolve.alias = {
       ...config.resolve.alias,
-      "@tinacms/mdx$": TINACMS_MDX_SHIM,
+      "@tinacms/mdx$": TINACMS_MDX_SHIM_ABS,
     };
 
     return config;
@@ -101,7 +106,7 @@ const config = {
     // Mirrors the webpack alias above — `next dev` uses Turbopack, `next build`
     // uses webpack, so both need it or dev and prod diverge.
     resolveAlias: {
-      "@tinacms/mdx": TINACMS_MDX_SHIM,
+      "@tinacms/mdx": TINACMS_MDX_SHIM_REL,
     },
   },
   experimental: {

